product_configurator/models/product.py

```diff
@@ -323,7 +323,7 @@ def check_config_user_access(self):
         )
         user_root = self.env.ref("base.user_root")
         user_admin = self.env.ref("base.user_admin")
-        if config_manager or self.env.user.id in [user_root.id, user_admin.id]:
+        if config_manager or self.env.user.id in [user_root.id, user_admin.id] or self.env.su:
```

Comment:
We need to test if self.env.su superseeds the previous check `self.env.user.id in [user_root.id, user_admin.id]` if it does then we should remove it because it will be redundant. Will check and let you know.

Reply:
@BenjaHe by default Odoo checks uid == SUPERUSER_ID and if it's true it sets env.su = True. Do we have an exception case somewhere because if we do not then the case is already covered in the condition.

Reply:
@PCatinean I made a little customization where I need to access specific product.config.session with a user in sudo mode. In this case the su flag is True but the uid is still the ID of the user

Reply:
Understood, I guess since it is possible to set just the flag itself it's a legitimate case, merging.